src/blocks.js

```diff
@@ -9193,9 +9193,24 @@ InputSlotMorph.prototype.menuFromDict = async function (
     }
 
     if (choices instanceof Function) {
-        choices = await choices.call(this);
+        if (!Process.prototype.enableJS) {
+            menu.addItem('JavaScript extensions for Snap!\nare turned off');
+            return menu;
+        }
+        choices = choices.call(this);
     } else if (isString(choices)) {
-        choices = await this[choices](enableKeyboard);
+        if (choices.indexOf('ext_') === 0) {
+            selector = choices.slice(4);
+            choices = SnapExtensions.menus.get(selector);
+            if (choices) {
+                choices = choices.call(this);
+            } else {
+                menu.addItem('cannot find extension menu "' + selector + '"');
+                return menu;
+            }
+        } else {
+            choices = this[choices]();
```

Comment:
Shouldn't this have an `await` here? It appears that we are losing support for async menus